### PR TITLE
Add support for csp_clear decorator.

### LINKED ISSUE #201: Support clearing/unsetting directives via decorator
Our application is served within an iframe. All but the initial authentication views need to have a `frame-ancestors` keyed to the tenants domain. We use a separate middleware to interact with the `_csp_update` API. To poke a hole for these authentication views, I've created a `csp_clear` decorator that clears the `frame_ancestors` directive.

I'm unsure if this has utility outside of our particular need. I'll be posting the PR shortly for you to review and decide if it's helpful or not.

### PR DESCRIPTION
**Pull request by tim-schilling**
This allows a directive to be cleared / unset entirely for a particular view.

Closes #201

### CODE REVIEW

File: docs/decorators.rst [hunk 1]
```diff
@@ -82,6 +82,28 @@ The arguments and values are the same as ``@csp_update``::
         return render(...)
 
 
+``@csp_clear``
+================
+
+The ``@csp_clear`` decorator allows you to **clear** a CSP directive. Clearing
+a directive will take priority over any of the other options.
```
Comment (stevejalim) [inline docs/decorators.rst:89]: I wonder if an admonition directive here like `warning` or even `danger` might be worthwhile.

File: docs/decorators.rst [hunk 2]
```diff
@@ -82,6 +82,28 @@ The arguments and values are the same as ``@csp_update``::
...
+The values to the decorator are the same as the :ref:`settings
+<configuration-chapter>` without the ``CSP_`` prefix, e.g. ``IMG_SRC``.
+(They are also case-insensitive.)::
+
+    from csp.decorators import csp_clear
+
+    # By default only allow responses to be served in an iframe
+    # from imgsrv.com
+    # settings.CSP_FRAME_ANCESTORS = ['imgsrv.com']
+    # Will allow this particular view to be served in an iframe from any\
```
Comment (stevejalim) [inline docs/decorators.rst:100]: Suggestion: adding a splash of extra clarification

```suggestion
    # 
    # Using the @csp_clear decorator to clear the `frame-ancestors` directive 
    # for this particular view will allow it to be served in an iframe from _any_
```

File: docs/decorators.rst [hunk 3]
```diff
@@ -82,6 +82,28 @@ The arguments and values are the same as ``@csp_update``::
...
 
+``@csp_clear``
+================
+
+The ``@csp_clear`` decorator allows you to **clear** a CSP directive. Clearing
+a directive will take priority over any of the other options.
+
+The values to the decorator are the same as the :ref:`settings
+<configuration-chapter>` without the ``CSP_`` prefix, e.g. ``IMG_SRC``.
+(They are also case-insensitive.)::
```
Comment (stevejalim) [inline docs/decorators.rst:93]: Suggestion: worth adding an explicit example of this

```suggestion
(They are also case-insensitive, so `img-src` also works)::
```

**Review by stevejalim — CHANGES_REQUESTED** [review]
Thanks for this @tim-schilling -- in my capacity as not-the-official-maintainer-but-keen-to-help, I can see how this could be useful to others -- and certainly makes it easier to get out of a hole if, say, auth has to happen on a different domain compared to the iframed content.

I've suggested some minor docs tweaks, and then it'll need rebasing and conflicts fixing, I'm afraid.

I'm thinking that if a 3.8 release remains focused on Django+Python updates and generally bringing the project up to date, this changeset could be a good one for 3.9

**Comment by robhudson** [comment]
Since the underlying code changed drastically I'm going to close this. We're discussing the need for this in #201.

**Pull request closed without merging**